packages/wrangler/src/__tests__/deploy.test.ts

```diff
@@ -4446,6 +4446,102 @@ addEventListener('fetch', event => {});`
 			`);
 		});
 
+		it("should warn when using smart placement with assets-first", async () => {
+			const assets = [
+				{ filePath: ".assetsignore", content: "*.bak\nsub-dir" },
+				{ filePath: "file-1.txt", content: "Content of file-1" },
+				{ filePath: "file-2.bak", content: "Content of file-2" },
+				{ filePath: "file-3.txt", content: "Content of file-3" },
+				{ filePath: "sub-dir/file-4.bak", content: "Content of file-4" },
+				{ filePath: "sub-dir/file-5.txt", content: "Content of file-5" },
+			];
+			writeAssets(assets, "assets");
+			writeWranglerConfig({
+				assets: {
+					directory: "assets",
+					experimental_serve_directly: true,
+				},
+				placement: {
+					mode: "smart",
+				},
+			});
+			const bodies: AssetManifest[] = [];
+			await mockAUSRequest(bodies);
+			mockSubDomainRequest();
+			mockUploadWorkerRequest({
+				expectedAssets: {
+					jwt: "<<aus-completion-token>>",
+					config: {
+						serve_directly: true,
+					},
+				},
+				expectedType: "none",
+			});
+
+			await runWrangler("deploy");
+
+			expect(std.warn).toMatchInlineSnapshot(`
+				"[33m▲ [43;33m[[43;30mWARNING[43;33m][0m [1mUsing assets with smart placement enabled may result in poor performance.[0m
```

Comment:
???

Reply:
This is just the rendering of the warning message. Nasty business 😆 
https://github.com/search?q=repo%3Acloudflare%2Fworkers-sdk%20%5B33m%E2%96%B2%20%1B%5B43%3B33m%5B%1B%5B43%3B30mWARNING%1B%5B43%3B33m%5D&type=code